packages/flutter/lib/src/semantics/binding.dart

```diff
@@ -28,12 +33,124 @@ mixin SemanticsBinding on BindingBase {
   static SemanticsBinding get instance => BindingBase.checkInstance(_instance);
   static SemanticsBinding? _instance;
 
+  /// Whether semantics information must be collected.
+  ///
+  /// Returns true if either the platform has requested semantics information
+  /// to be generated or if [ensureSemantics] has been called otherwise.
+  ///
+  /// To get notified when this value changes register a listener with
+  /// [addSemanticsEnabledListener].
+  bool get semanticsEnabled {
+    assert(_semanticsEnabled.value == (_outstandingHandles > 0));
+    return _semanticsEnabled.value;
+  }
+  late final ValueNotifier<bool> _semanticsEnabled = ValueNotifier<bool>(platformDispatcher.semanticsEnabled);
+
+  /// Adds a `listener` to be called when [semanticsEnabled] changes.
+  ///
+  /// See also:
+  ///
+  ///  * [removeSemanticsEnabledListener] to remove the listener again.
+  ///  * [ValueNotifier.addListener], which documents how and when listeners are
+  ///    called.
+  void addSemanticsEnabledListener(VoidCallback listener) {
+    _semanticsEnabled.addListener(listener);
+  }
+
+  /// Removes a `listener` added by [addSemanticsEnabledListener].
+  ///
+  /// See also:
+  ///
+  ///  * [ValueNotifier.removeListener], which documents how listeners are
+  ///    removed.
+  void removeSemanticsEnabledListener(VoidCallback listener) {
+    _semanticsEnabled.removeListener(listener);
+  }
+
+  int _outstandingHandles = 0;
+
+  /// Creates a new [SemanticsHandle] and requests the collection of semantics
+  /// information.
+  ///
+  /// Semantics information are only collected when there are clients interested
+  /// in them. These clients express their interest by holding a
+  /// [SemanticsHandle].
+  ///
+  /// Clients can close their [SemanticsHandle] by calling
+  /// [SemanticsHandle.dispose]. Once all outstanding [SemanticsHandle] objects
+  /// are closed, semantics information are no longer collected.
+  SemanticsHandle ensureSemantics() {
```

Comment:
What is the use case to expose this API?

Reply:
Rephrasing from the PR description:

Clients can use ensureSemantics to turn on semantics globally for the entire app. Previously, this was all handled internally by the PipelineOwner. However, in the multi view world we will have multiple PipelineOwners (one to power each view), so a single source of truth to turn on semantics for of all these PipelineOwners across the entire app is needed.